[docs] Add Menu component example with explicit positioning prop values (#23167)
mui-org/material-ui: docs/src/pages/components/menus/menus.md

waiAria: https://www.w3.org/TR/wai-aria-practices/#menubutton
---

# Menu

<p class="description">Menus display a list of choices on temporary surfaces.</p>

> A Menu displays a list of choices on a temporary surface. It appears when the user interacts with a button, or other control.

Reviewer: ```suggestion
A menu displays a list of choices on a temporary surface. It appears when the user interacts with a button, or other control.
```

{{"component": "modules/components/ComponentLinkHeader.js"}}

## Basic menu

> Simple menus open over the anchor element by default (this option can be [changed](#positioned-menus) via props). When close to a screen edge, simple menus vertically realign to make sure that all menu items are completely visible.

Reviewer: ```suggestion
A basic menu open over the anchor element by default (this option can be [changed](#positioned-menus) via props). When close to a screen edge, a basic menu vertically realigns to make sure that all menu items are completely visible.
```

> A basic menu open over the anchor element by default (this option can be [changed](#positioned-menus) via props). When close to a screen edge, a basic menu vertically realigns to make sure that all menu items are completely visible.

Author: ```suggestion
A basic menu opens over the anchor element by default (this option can be [changed](#positioned-menus) via props). When close to a screen edge, a basic menu vertically realigns to make sure that all menu items are completely visible.
```

> A basic menu opens over the anchor element by default (this option can be [changed](#positioned-menus) via props). When close to a screen edge, a basic menu vertically realigns to make sure that all menu items are completely visible.

Reviewer 2: ```suggestion
A basic menu opens over the anchor element by default (this option can be [changed](#menu-positioning) via props). When close to a screen edge, a basic menu vertically realigns to make sure that all menu items are completely visible.
```

{{"demo": "pages/components/menus/SimpleListMenu.js"}}

> ## Positioned menus

Reviewer 2: @mbrookes It seems that we are not very consistent in the documentation about the plural. For instance:

- `## Customized tooltips`
- `## Customized accordion`
- `## Positioned Tooltips`
- `## Positioned Popper`
- `### Transition`
- `## Transitions`
- `## Basic alerts`
- `## Basic rating`

I'm not sure what we should do

Reviewer: Singular works here. I also wouldn't shy away from switching up the word order where it makes more sense, so: "Menu position"?

Reviewer 2: I guess we could update all the headers at once, that would help maximize consistency.

Author: > I guess we could update all the headers at once, that would help maximize consistency.

@oliviertassinari I can singularize the headers for this file as part of this PR if you'd like

Because the `Menu` component uses the `Popover` component to position itself, you can use the same [positioning props](/components/popover/#anchor-playground) to position it.

> 

Reviewer: ```suggestion
```
